packages/core/strapi/lib/services/entity-service/types/index.d.ts

```diff
@@ -1,6 +1,20 @@
+import type { Attribute, Common, EntityService } from '@strapi/strapi';
+
 // TODO: Move params to @strapi/utils (related to convert-query-params)
 export * as Params from './params';
 
 export * from './plugin';
 
-export interface EntityService {}
+type Pagination = {
+  page: number;
+  pageSize: number;
+  pageCount: number;
+  total: number;
+};
+
+export interface EntityService {
+  findPage<TContentTypeUID extends Common.UID.ContentType>(
+    uid: TContentTypeUID,
+    params?: EntityService.Params.Read<TContentTypeUID>
+  ): Promise<{ results: Attribute.GetValues<TContentTypeUID>[]; pagination: Pagination } | null>;
```

Comment:
Can we actually get `null` here though?
My guess would be that `results` is an empty array if the queries don't find anything